ruma-identity-service-api/src/keys/valid_public_key/v2.rs

```diff
@@ -0,0 +1,56 @@
+//! [GET /_matrix/identity/v2/pubkey/isvalid](https://matrix.org/docs/spec/identity_service/latest#get-matrix-identity-v2-pubkey-isvalid)
+
+use ruma_api::ruma_api;
+
+ruma_api! {
+    metadata: {
+        description: "Check whether a long-term public key is valid. The response should always be the same, provided the key exists.",
+        method: GET,
+        name: "valid_public_key",
+        path: "/_matrix/identity/v2/pubkey/isvalid",
+        authentication: None,
+        rate_limited: false,
+    }
+
+    #[derive(Default)]
+    request: {}
+
+    response: {
+        /// Whether the public key is recognised and is currently valid.
+        pub valid: bool,
+    }
+}
+
+impl Request {
+    /// Create an empty `Request`.
+    ///
+    /// Note: client must supply public key query parameter.
+    ///
+    /// Example:
+    ///
+    /// ```rust-no-build
```

Comment:
It would be nice if this was `no-run` but still checked as compiling. I think the only changes necessary would be to remove `ruma::` from the two imports that have it and add `ruma_client` as a dev-dependency to this crate.

Reply:
I removed the docs, since they're kind of unnecessary now. Should I add them back?

Reply:
Yeah, I guess they aren't needed. Really should have more docs on ruma-client instead.